File: Dynamic Programming/Java/2747 피보나치 수.java
import java.util.Scanner;

public class Main {
    public static void main(String args[]) {
      Scanner s = new Scanner(System.in);
      int n = s.nextInt();
      int[] dp = new int[46];
      
      dp[0] = 0;
      dp[1] = 1;
      dp[2] = 1;
      for(int i = 3; i <= n; i++) {
          dp[i] = dp[i-2] + dp[i-1];
      }
      
      System.out.format("%d", dp[n]);
    }
}

// http://boj.kr/e1bb61e001614488b91dfaac404097b3

File: Beakjoon/_etc/Kotlin/2747 피보나치 수.kt

import java.util.Scanner

fun main(args: Array<String>) {
    val sc: Scanner = Scanner(System.`in`)
    val n = sc.nextInt()
    var dp = Array(46){0}
    
    dp[0] = 0
    dp[1] = 1
    dp[2] = 1
    
    for(i in 3..n) {
        dp[i] = dp[i-1] + dp[i-2]
    }
    
    println(dp[n])
}

// http://boj.kr/d3f7decf81014732b2a4fe951da4d910
